src/pages/ShufflePlayer.js

```diff
@@ -7,75 +7,73 @@ import Player from '../components/Player';
 import PlaylistSelector from '../components/PlaylistSelector';
 import PlayedHistory from '../components/PlayedHistory';
 import CurrentVideoInfo from '../components/CurrentVideoInfo';
+import useVideoHook from '../hooks/VideoHook';
 
-function ShufflePlayer(props) {
-  const [isLoaded, setIsLoaded] = useState(false);
+function randomInteger(min, max) {
+  return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function ShufflePlayer() {
   const [loadedPlaylists, setLoadedPlaylists] = useState([]);
-  const [loadedVideos, setLoadedVideos] = useState([]);
   const [currentVideo, setCurrentVideo] = useState({});
   const [playedVideos, setPlayedVideos] = useState([]);
   const [repeatVideo, setRepeatVideo] = useState(false);
+  
+  const [playlistIds, setPlaylistIds] = useState([]);
+  const [videoHookResult, reloadVideos] = useVideoHook(playlistIds);
 
   function loadPlaylists() {
     axios.get(AppConstants.APIEndpoints.TRACKED_PLAYLISTS)
     .then(response => {
       setLoadedPlaylists(response.data);
-      setIsLoaded(true);
     })
     .catch(error => console.log(`Couldn't retrieve tracked playlists! ${error}`))
   }
 
-  function loadVideos() {
-    const playlistIdsToLoad = loadedPlaylists
-      .filter(p => p.is_default)
-      .map(p => p.playlist_id);
-    const requestBody = { playlist_ids: playlistIdsToLoad };
-    axios.post(AppConstants.APIEndpoints.SHUFFLE, requestBody)
-    .then(response => setLoadedVideos(response.data.songs))
-    .catch(error => console.log(`Couldn't retrieve playlist videos! ${error}`))
-  }
-
   function togglePlaylistSelection(togglePlaylistId) {
     const toggledOnePlaylist = loadedPlaylists.map(p => { 
       return p.playlist_id === togglePlaylistId ? 
         { ...p, is_default: !p.is_default } : p});
+    const playlistIds = toggledOnePlaylist
+      .filter(p => p.is_default)
+      .map(p => p.playlist_id);
+    
+    setPlaylistIds(playlistIds);
     setLoadedPlaylists(toggledOnePlaylist);
   }
 
   function onSelectNone() {
     const selectedNoPlaylists = loadedPlaylists.map(p => { 
       return { ...p, is_default: false }});
+    setPlaylistIds("");
```

Comment:
Should be an empty array, not empty string here?

Reply:
Topic for a separate PR: I feel like the `is_default` concept should be separated into two things.
One indicates that it's a default selection when the app loads and the other is what is currently selected. I don't think calling what's currently selected `is_default` makes sense

Reply:
updated. Yeah its just a carry over from the api, it should be a separate concept that we can set from the default value from the api on start